Makefile

```diff
@@ -1,20 +1,28 @@
-
 KERNEL_VERSIONS = 2013.4 2016.4
 
-drivers = intc sarspi uio
+drivers = intc sarspi uio allocated-gpio
 
 default: 2016.4
 
 all: $(KERNEL_VERSIONS)
 
 201%: KVER=$@
+201%:	KREL=$(shell cat $(KDIR)/include/config/kernel.release)
 201%:
 	./version.sh
 	$(foreach dir,$(drivers),$(MAKE) -C $(dir) all;)
-	mkdir -p build/$@
-	$(foreach dir,$(drivers),mv $(dir)/*.ko build/$@;)
-	$(foreach dir,$(drivers),cp $(dir)/*rules build/;)
+	mkdir -p build/udev/
+	cp */*rules build/udev/
+	mkdir -p build/$@/usr/lib/modules/$(KREL)/kernel/imsar
+	rsync -a $(KDIR)/arch/arm/boot/uImage build/$@/
```

Comment:
It looks like this is doing more than just driver building. It also copies the uImage as well?

Reply:
Yes.  That makes sure you can get the uImage that the modules were build against, and makes this jenkins job a one-stop-shop for all things kernel.